src/main/java/com/elixr/poc/rest/controller/PurchaseModificationController.java

```diff
@@ -0,0 +1,36 @@
+package com.elixr.poc.rest.controller;
+
+import com.elixr.poc.data.Purchase;
+import com.elixr.poc.exception.NoRecordFoundException;
+import com.elixr.poc.service.PurchaseService;
+import jakarta.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@Validated
+@RequestMapping("/controller")
+public class PurchaseModificationController {
+    private final PurchaseService purchaseService;
+
+    public PurchaseModificationController(PurchaseService purchaseService) {
+        this.purchaseService = purchaseService;
+    }
+
+    /**
+     * Calling updatePurchase method with the parameter purchaseId to update the purchase by purchaseId.
+     * And handling the Exception if the userId is not matching.
+     *
+     * @param purchaseId
+     * @return
+     * @throws NoRecordFoundException
+     */
+    @PatchMapping("/purchase/{purchaseId}")
+    public ResponseEntity<Purchase> updateUserPartially(@PathVariable("purchaseId") UUID purchaseId, @RequestBody @Valid Purchase purchaseDetails) throws NoRecordFoundException {
```

Comment:
Dont use Purchase class in argument or return type of a Controller.
use PurchaseReuqest & PurchaseResponse. 
do the conversion from request/response object to Purchase object in controller

Reply:
Updated.